shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformChannel.java

```diff
@@ -431,6 +453,39 @@ private SystemChromeStyle decodeSystemChromeStyle(@NonNull JSONObject encodedSty
    * PlatformChannel#setPlatformMessageHandler(PlatformMessageHandler)}.
    */
   public interface PlatformMessageHandler {
+    /**
+     * A data class for [PlatformMessageHandler.setPreferredOrientations] and
+     * [PlatformMessageHandler.getPreferredOrientations].
+     */
+    public class AndroidOrientation {
+      /** Holds a value from ActivityInfo.SCREEN_ORIENTATION constants. */
+      private int androidOrientation;
+      /**
+       * Holds a value that was requested by the Flutter application. It is needed due to the fact
+       * that ActivityInfo.SCREEN_ORIENTATION constants can not describe all combinations of screen
+       * orientations that flutter application could pass: 4 elements from the Flutter's
+       * DeviceOrientation enum give a total of 16 possible combinations and not all of them could
+       * be mapped to a ActivityInfo.SCREEN_ORIENTATION.
+       */
+      private int flutterRequestedOrientation;
+
+      public AndroidOrientation(
+          @NonNull int androidOrientation, @NonNull int flutterRequestedOrientation) {
+        this.androidOrientation = androidOrientation;
+        this.flutterRequestedOrientation = flutterRequestedOrientation;
```

Comment:
It is a good idea, but I'm not sure where is the best place for it:
- in this repo in the `decodeOrientations` method,
- or in the main flutter repo in the `setPreferredOrientations`.

I think the second option is more useful for developers, but it might be an issue to keep that log message updated.

Also, I found the following [issue](https://github.com/flutter/flutter/issues/99149) which tracks the problem.

Reply:
I can see updating setPreferredOrientations also being valuable but in this PR I think warning that the decode found something it didnt expect even if the passed value has an issue someplace else. 

Since you agree lets add a warning in `decodeOrientations`.